packages/wdio-types/src/Capabilities.ts

```diff
@@ -817,27 +817,398 @@ export interface IECapabilities {
+     */
+    enableAnimations?: boolean
+
+    /**
+     * Delays system alerts, such as alerts asking for permission to access the camera, to prevent app crashes at startup. 
+     * resigningEnabled needs to be enabled if this is set to true.
+     * 
+     * iOS Real Devices only.
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#systemalertsdelayenabled
+     */
+    systemAlertsDelayEnabled?: boolean
+
+    /**
+     * Specify the amount of time (in milliseconds) that the test should be allowed to find and assign an available device before the test will fail. 
+     * The default value is 900000 milliseconds (15 minutes) and the max is 1800000 milliseconds (30 minutes).
+     * 
+     * Real Devices only.
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#sessioncreationtimeout 
+     */
+    sessionCreationTimeout?: boolean
+
+    /**
+     * Specify the amount of automatic retries that Sauce Labs will execute to find and assign an available device before the test will fail.
+     * The default value is 1 and the max is 3.
+     * 
+     * Real Devices only.
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#sessioncreationretry
+     */
+    sessionCreationRetry?: number
+
+    /**
+     * Desktop and Mobile Capabilities: Sauce-Specific – Optional
+     * see https://docs.saucelabs.com/dev/test-configuration-options/#desktop-and-mobile-capabilities-sauce-specific--optional
+     */
+
     /**
      * Used to record test names for jobs and make it easier to find individual tests.
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#name
      */
     name?: string
+
     /**
      * Used to associate jobs with a build number or app version, which is then displayed
      * on both the Dashboard and Archives view
      *
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#build
      * @example `build-1234`
      */
-    build?: string | number
+    build?: string
+
     /**
      * User-defined tags for grouping and filtering jobs in the Dashboard and Archives view.
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#tags
      * @example
      * ```
      * ["tag1", "tag2", "tag3"]
      * ```
      */
     tags?: string[]
+
+    /**
+     * Sets your Sauce Labs username for a test.
+     * 
+     * You can either set "username" in capabilities or specify it in the Sauce URL as Basic Authentication. For Visual Tests), this must be set in capabilities.
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#username
+     */
+    username?: string
+
+    /**
+     * Sets your Sauce Labs access key for the test.
+     * 
+     * You can either set "accessKey" in capabilities or specify it in the Sauce URL as Basic Authentication. For Visual Tests, this must be set in capabilities.
+     * @see https://docs.saucelabs.com/dev/test-configuration-options/#accesskey
+     */
+    accessKey?: string
```

Comment:
Do we want to recommend this? It could potentially expose Sauce credentials in test logs since WebdriverIO does nothing to obfuscate this

Reply:
We could add a callout to this possibility - but the second part is more important:
> For Visual Tests, this must be set in capabilities.

Reply:
FWIW Lambdatest and Browserstack also have username and access key as part of their capabilities.

Reply:
I will leave it up to you, IMHO this shouldn't be recommended to users.